Rendera/include/scene.hpp

```diff
@@ -85,29 +90,34 @@ class Scene{
 
         }
         ~Scene(){
-            delete hitable_sphere;
             delete hitinfo;
         }
-        void set_sphere(int index, vec3 center, float radius, Material mat){
-            if (index < n_sphere)
-                hitable_sphere[index] = sphere(center, radius, mat);
-            else
-                std::cout<<"accessing wrong index"<<std::endl;
+        void add_sphere(vec3 center, float radius, Material mat){
+            n_sphere++;
+            hitable_sphere.push_back(Sphere(center, radius, mat));
+        }
+        void add_cube(vec3 position, vec3 a, vec3 b, vec3 c, Material mat){
+            n_cube++;
+            hitable_cube.push_back(Cube(position, a, b, c, mat));
         }
 
 
 
         void hit_object(int x, int y){
-            hitinfo[y*img_width + x].ray_ = ray(cam.get_ray(x, y));
-            for(int i = 0; i < n_sphere; i++){
+            std::vector<Sphere>::iterator sp;
+            std::vector<Cube>::iterator cb;
+            hitinfo[y*img_width + x].ray_ = cam.get_ray(x, y);
+            // std::cout<<cam.get_ray(x, y).get_direction();
+            for(sp = hitable_sphere.begin(); sp != hitable_sphere.end(); ++sp){
                 Intersection i_;
-                bool hits = i_.hit(hitable_sphere[i], hitinfo[y*img_width + x].ray_);
+                bool hits = i_.hit(*sp, hitinfo[y*img_width + x].ray_);
                 if (hitinfo[y*img_width + x].ishitting && hits){
                     if (hitinfo[y*img_width + x].dist > i_.dist_1()){
                         hitinfo[y*img_width + x].dist = i_.dist_1();
                         hitinfo[y*img_width + x].intersection_pt = hitinfo[y*img_width + x].ray_.fetch(hitinfo[y*img_width + x].dist); 
-                        hitinfo[y*img_width + x].normal = hitable_sphere[i].normal(hitinfo[y*img_width + x].intersection_pt);
-                        hitinfo[y*img_width + x].shade = lighting(hitable_sphere[i].mat(), hitinfo[y*img_width + x].ray_, hitinfo[y*img_width + x].normal, hitinfo[y*img_width + x].intersection_pt, light_src, light_col);
+                        hitinfo[y*img_width + x].normal = (*sp).normal(hitinfo[y*img_width + x].intersection_pt);
+                        hitinfo[y*img_width + x].shade = lighting((*sp).mat(), hitinfo[y*img_width + x].ray_, hitinfo[y*img_width + x].normal, hitinfo[y*img_width + x].intersection_pt, light_src, light_col);
+                        hitinfo[y*img_width + x].shade += lighting((*sp).mat(), hitinfo[y*img_width + x].ray_, hitinfo[y*img_width + x].normal, hitinfo[y*img_width + x].intersection_pt, light_src2, light_col);
```

Comment:
This code is a massive wall of text as it stands. Surely there is a way to make it easier on the eyes?